bpf/kmesh/workload/sockops_workload.c

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2024 The Kmesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <linux/bpf.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_endian.h>
+#include "bpf_log.h"
+#include "workload.h"
+
+#define AF_INET  (2)
+#define AF_INET6 (10)
+
+enum family_type {
+    IPV4,
+    IPV6,
+};
+
+struct ringbuf_msg_type {
+    __u32 type; 
+    struct bpf_sock_tuple tuple;
+};
+
+// return 1 if pod with skops.daddr is managed by Kmesh, else return 0
+static inline int dst_is_managed(struct bpf_sock_ops *skops)
+{
+    return 0;
+}
+
+static inline void build_auth_key(struct bpf_sock_ops *skops, struct bpf_sock_tuple* key)
+{
+    if(key) {
+        key->ipv4.saddr = skops->remote_ip4;
+        key->ipv4.sport = skops->remote_port >> 16;
+        key->ipv4.daddr = skops->local_ip4;
+        key->ipv4.dport = bpf_htonl(skops->local_port) >> 16;
+    }
+}
+
+// clean map_of_auth
+static inline void clean_auth_map(struct bpf_sock_ops *skops)
+{
+    struct bpf_sock_tuple tuple_key = {0};
+    build_auth_key(skops, &tuple_key);
+    long ret = bpf_map_delete_elem(&map_of_auth, &tuple_key);
+    if(ret && ret != -ENOENT) {
+        BPF_LOG(INFO, SOCKOPS, "map_of_auth bpf_map_delete_elem failed, ret: %d\n", ret);
+    }
+}
+
+SEC("sockops")
+int socket_migration(struct bpf_sock_ops *skops)
```

Comment:
As we discussed the name is not consistent with what it really does.

Reply:
Has been changed to ```record_tuple```